### PR TITLE
Wireframes

### PR DESCRIPTION
**Pull request by etienne-monsieurbiz**
### Features:

- Add possibility to add a `wireframe` key in the yaml declaration of the ui element
- Easily add wireframes file as svg.twig to add them in a project for custom wireframes
- Add wireframe svg in the list of ui element

Capture d’écran 2024-03-06 à 14 16 14

### CODE REVIEW

**Review by maximehuran — COMMENTED** [review]
What is the advantage to have SVGs instead of simple images ?

**Comment by etienne-monsieurbiz** [comment]
@maximehuran 

- The fact that I can modify the extension and render the file through twig
- Size of file that is most likely smaller than jpg, png, others
- I didn't do it but we could also update the svg.twig with attributes

File: src/UiElement/UiElementTrait.php [hunk 1]
```diff
@@ -23,6 +25,8 @@ trait UiElementTrait
 
     protected bool $ignored = false;
 
+    protected Environment $twig;
```
Comment (jacquesbh) [inline src/UiElement/UiElementTrait.php:28]: **❔ question:** ‏What about having this nullable?

File: src/UiElement/UiElementTrait.php [hunk 2]
```diff
@@ -87,6 +99,32 @@ public function getIcon(): string
...
+    public function getWireframeSvg(): string
+    {
+        try {
+            $code = $this->getWireframe();
+
+            if (empty($code)) {
+                return '';
+            }
+
+            return $this->twig->render(sprintf('@MonsieurBizSyliusRichEditorPlugin/Wireframe/%s.svg.twig', $code));
```
Comment (jacquesbh) [inline src/UiElement/UiElementTrait.php:122]: **🎯 suggestion:** If $twig is nullable, we can have an empty string returned. WDYT?

Also, what about testing if the template exists as well as `%s.html.twig` and `%s.svg.twig`?
This way we can override the SVG by using our own HTML, like an `img` or anything else!‏

File: src/UiElement/UiElementTrait.php [hunk 3]
```diff
@@ -87,6 +99,32 @@ public function getIcon(): string
...
+     */
+    public function getWireframe(): string
+    {
+        return $this->metadata->getParameter('wireframe');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getWireframeSvg(): string
```
Comment (jacquesbh) [inline src/UiElement/UiElementTrait.php:113]:
```suggestion
    public function getWireframeAsHtml(): string
```

**Review by jacquesbh — CHANGES_REQUESTED** [review]

File: src/UiElement/UiElementTrait.php (hunk 1 above)
Comment (etienne-monsieurbiz) [reply src/UiElement/UiElementTrait.php:28]: Why should it be nullable ?

**Pull request merged**